plugins/utils/which-key.nix

```diff
@@ -55,257 +265,308 @@ with lib;
       ];
     };
 
+    notify = defaultNullOpts.mkBool true "Show a warning when issues were detected with your mappings.";
+
+    triggers = defaultNullOpts.mkListOf (with types; attrsOf anything) [
+      {
+        __unkeyed = "<auto>";
+        mode = "nxsot";
+      }
+    ] "Manually setup triggers";
+
+    defer =
+      defaultNullOpts.mkLuaFn
+        (mkRaw ''
+          function(ctx)
+            return ctx.mode == "V" or ctx.mode == "<C-V>
+            end'')
+        ''
+          Start hidden and wait for a key to be pressed before showing the popup.
+
+          Only used by enabled xo mapping modes.
+        '';
+
     plugins = {
-      marks = helpers.defaultNullOpts.mkBool true "shows a list of your marks on ' and `";
+      marks = defaultNullOpts.mkBool true "Shows a list of your marks on ' and `";
 
-      registers = helpers.defaultNullOpts.mkBool true ''shows your registers on " in NORMAL or <C-r> in INSERT mode'';
+      registers = defaultNullOpts.mkBool true ''Shows your registers on " in NORMAL or <C-r> in INSERT mode'';
 
       spelling = {
-        enabled = helpers.defaultNullOpts.mkBool true "enabling this will show WhichKey when pressing z= to select spelling suggestions";
-        suggestions = helpers.defaultNullOpts.mkInt 20 "how many suggestions should be shown in the list?";
+        enabled = defaultNullOpts.mkBool true "Enabling this will show WhichKey when pressing z= to select spelling suggestions";
```

Comment:
```suggestion
        enabled = defaultNullOpts.mkBool true ''
          Enabling this will show WhichKey when pressing `z=` to select spelling suggestions.
        '';
```